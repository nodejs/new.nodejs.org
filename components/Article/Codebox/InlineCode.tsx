import styles from './index.module.scss';
import type { FC, PropsWithChildren } from 'react';

const InlineCode: FC<PropsWithChildren> = ({ children }) => (
  <code className={styles.code}>{children}</code>
);

export default InlineCode;
