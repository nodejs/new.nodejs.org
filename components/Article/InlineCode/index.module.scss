@use 'styles/mixins/components';

.code {
  @include components.code;

  font-weight: var(--font-weight-light);
  padding: 0 6px;

  white-space: break-spaces;
}

[data-theme='light'] {
  .code {
    background-color: var(--black2);
    color: var(--black9);
  }

  a .code {
    background-color: transparent;
    color: var(--color-text-accent);
    padding: 0;
  }
}

[data-theme='dark'] {
  .code {
    background-color: var(--black9);
    color: var(--black2);
  }

  a .code {
    background-color: transparent;
    color: var(--color-text-accent);
    padding: 0;
  }
}
