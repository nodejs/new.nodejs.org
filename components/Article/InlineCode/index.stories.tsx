import InlineCode from './index';
import type { FC } from 'react';
import type { Meta as MetaObj, StoryObj } from '@storybook/react';

type DecoratedInlineCodeProps = { code: string };

const DecoratedInlineCode: FC<DecoratedInlineCodeProps> = ({ code }) => (
  <InlineCode>
    <code>{code}</code>
  </InlineCode>
);

type Story = StoryObj<typeof DecoratedInlineCode>;
type Meta = MetaObj<typeof DecoratedInlineCode>;

const code = 'const a = 1;';

export const Default: Story = { args: { code } };

export default { component: DecoratedInlineCode } as Meta;
