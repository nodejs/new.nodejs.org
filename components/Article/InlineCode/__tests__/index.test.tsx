import { render } from '@testing-library/react';
import InlineCode from '../index';

describe('InlineCode component', () => {
  it('should render correctly', () => {
    const { container } = render(
      <InlineCode>
        <code>const a = 1;</code>
      </InlineCode>
    );

    expect(container).toMatchSnapshot();
  });
});
