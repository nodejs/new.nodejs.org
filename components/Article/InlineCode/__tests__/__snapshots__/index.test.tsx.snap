// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InlineCode component should render correctly 1`] = `
<div>
  <code
    class="code"
  >
    <code>
      const a = 1;
    </code>
  </code>
</div>
`;
